contracts/bridge/contracts/common/message/InboundLane.sol

```diff
@@ -0,0 +1,285 @@
+        }
+    }
+
+    function relayers_back() public view returns (address pre_relayer) {
+        if (relayers_size() > 0) {
+            uint256 back = relayersRange.back;
+            pre_relayer = relayers[back].relayer;
+        }
+    }
+
+	// Get lane data from the storage.
+    function data() public view returns (InboundLaneData memory lane_data) {
+        uint256 size = relayers_size();
+        if (size > 0) {
+            lane_data.relayers = new UnrewardedRelayer[](size);
+            uint256 front = relayersRange.front;
+            for (uint256 index = 0; index < size; index++) {
+                lane_data.relayers[index] = relayers[front + index];
+            }
+        }
+        lane_data.last_confirmed_nonce = inboundLaneNonce.last_confirmed_nonce;
+        lane_data.last_delivered_nonce = inboundLaneNonce.last_delivered_nonce;
+    }
+
+    // storage proof issue: must use latest commitment in lightclient, cause we rm mmr root
+    function commit() public returns (bytes32) {
+        commitment = hash(data());
+        return commitment;
+    }
+
+    /* Private Functions */
+
+    // Receive state of the corresponding outbound lane.
+    function receive_state_update(uint256 latest_received_nonce) internal returns (uint256) {
+        uint256 last_delivered_nonce = inboundLaneNonce.last_delivered_nonce;
+        uint256 last_confirmed_nonce = inboundLaneNonce.last_confirmed_nonce;
+        require(latest_received_nonce <= last_delivered_nonce, "Lane: InvalidReceivedNonce");
+        if (latest_received_nonce > last_confirmed_nonce) {
+            uint256 new_confirmed_nonce = latest_received_nonce;
+            uint256 front = relayersRange.front;
+            uint256 back = relayersRange.back;
+            for (uint256 index = front; index <= back; index++) {
+                UnrewardedRelayer storage entry = relayers[index];
+                if (entry.messages.end <= new_confirmed_nonce) {
+                    // Firstly, remove all of the records where higher nonce <= new confirmed nonce
+                    delete relayers[index];
+                    relayersRange.front = index + 1;
+                } else if (entry.messages.begin < new_confirmed_nonce) {
+                    // Secondly, update the next record with lower nonce equal to new confirmed nonce if needed.
+                    // Note: There will be max. 1 record to update as we don't allow messages from relayers to
+                    // overlap.
+                    entry.messages.dispatch_results >>= (new_confirmed_nonce + 1 - entry.messages.begin);
+                    entry.messages.begin = new_confirmed_nonce + 1;
+                }
+            }
+            inboundLaneNonce.last_confirmed_nonce = new_confirmed_nonce;
+        }
+        return latest_received_nonce;
+    }
+
+    // Receive new message.
+    function receive_message(Message[] memory messages) internal returns (uint256 dispatch_results) {
```

Comment:
```suggestion
    function dispatch_message(Message[] memory messages) internal returns (uint256 dispatch_results) {
```

Reply:
This function name is consistent with the s2s framework, to make it easier to review.

Reply:
ok